aerogear/mobile-developer-console
# UPS Operator integration

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9435

## What
- Added ability to customise the resource management for each of the resource type
- Added the PushVariantCR object
- Added the PushApplicationCR object
- Customised the behaviour of the PushVariantCR Manager so that variants are managed

## How to test

1. Run the MDC as usual
2. Install the UPS operator in the `mobile-console` namespace
3. Use the UPS operator to create a `UnifiedPush` instance
4. Create an app in MDC
5. Click to the app details
6. Click on mobile services
7. Try to bind and unbind the Push Service. Note that Push Service can be bound 2 times, one for android and one for iOS

## Tasks:
- [x] Add ability to bind an android variant
- [x] Add ability to bind an ios variant 
- [x] Add ability to unbind the variants
- [x] Customise the watchers to be able to support services using more than one custom resource
- [x] Add ability to automatically create the UPS application when creating a variant if it doesn't already exists


## KNOWN BUGS

There is a bug in the binding window: if you select ios and then try to move back to android, the fields does not gets updated. This will be fixed as part of another issue.
